reporters/kamon-opentelemetry/src/main/scala/kamon/otel/MetricsConverter.scala

```diff
@@ -0,0 +1,158 @@
+  private val toNs = to.toEpochMilli * 1000000
+
+  private def instrumentationScopeInfo(snapshot: MetricSnapshot[_, _]): InstrumentationScopeInfo =
+    InstrumentationScopeInfo.create("kamon-metrics", kamonVersion, null)
+
+  private def toString(unit: MeasurementUnit): String = unit.magnitude.name
+
+  private def toGaugeDatum(g: Snapshot[Double]): DoublePointData = ImmutableDoublePointData.create(fromNs, toNs, SpanConverter.toAttributes(g.tags), g.value)
+
+  private def toGaugeData(g: Seq[Snapshot[Double]]): GaugeData[DoublePointData] = ImmutableGaugeData.create(g.map(toGaugeDatum).asJava)
+
+  def convertGauge(gauge: MetricSnapshot.Values[Double]): MetricData =
+    ImmutableMetricData.createDoubleGauge(
+      resource,
+      instrumentationScopeInfo(gauge),
+      gauge.name,
+      gauge.description,
+      toString(gauge.settings.unit),
+      toGaugeData(gauge.instruments))
+
+  private def toExplicitHistogramDatum(s: Snapshot[Distribution]): HistogramPointData = {
+    val boundaries = ArrayBuffer.newBuilder[JDouble]
+    val counts = ArrayBuffer.newBuilder[JLong]
+    for (el <- s.value.bucketsIterator) {
+      counts += el.frequency
+      boundaries += el.value.toDouble
+    }
+    ImmutableHistogramPointData.create(
+      fromNs,
+      toNs,
+      SpanConverter.toAttributes(s.tags),
+      JDouble valueOf s.value.sum.toDouble,
+      JDouble valueOf s.value.min.toDouble,
+      JDouble valueOf s.value.max.toDouble,
+      boundaries.result().dropRight(1).asJava,
+      counts.result().asJava
+    )
+  }
+
+  private def toExplicitHistogramData(distributions: Seq[Snapshot[Distribution]]): Option[HistogramData] =
+    distributions.filter(_.value.buckets.nonEmpty) match {
+      case Nil => None
+      case nonEmpty => Some(ImmutableHistogramData.create(AggregationTemporality.DELTA, nonEmpty.map(toExplicitHistogramDatum).asJava))
+    }
+
+  def convertExplicitHistogram(histogram: MetricSnapshot.Distributions): Option[MetricData] =
+    toExplicitHistogramData(histogram.instruments).map(d =>
+      ImmutableMetricData.createDoubleHistogram(
+        resource,
+        instrumentationScopeInfo(histogram),
+        histogram.name,
+        histogram.description,
+        toString(histogram.settings.unit),
+        d))
+
+  private def toExponentialHistogramData(distributions: Seq[Snapshot[Distribution]]): Option[ExponentialHistogramData] =
+    distributions.filter(_.value.buckets.nonEmpty) match {
+      case Nil => None
+      case nonEmpty =>
+        val mapped = nonEmpty.flatMap { s =>
+          s.value match {
+            case zigZag: Distribution.ZigZagCounts =>
+              logger.error("Unable to construct exponential histogram data - Unimplemented")
+              None
+//              Some(ExponentialHistogramPointData.create(
+//                ???, zigZag.sum, ???, ???, ???, fromNs, toNs, SpanConverter.toAttributes(s.tags), new JArrayList[DoubleExemplarData]()
+//              ))
```

Comment:
Not got around to this yet...

Reply:
I think you probably don't need to understand anything about the ZigZagCounts' implementation to get this going. You will probably need to add a setting for the exponential histogram scale and then go through Kamon's bucketsIterator and map them to the appropriate bucket on the exponential histogram. Sounds weird, but once you get to it you will see it :)

Reply:
Ok, I get how that'd work now! Makes sense